gematria/datasets/exegesis_benchmark.cc

```diff
@@ -90,9 +96,11 @@ Expected<BenchmarkCode> parseJSONBlock(
 
   if (!RegisterDefinitions)
     return llvm::make_error<StringError>(
-        errc::invalid_argument,
-        "Malformed basic block: no register definitions array");
+        errc::invalid_argument, "Malformed basic block: Basic block at index " +
+                                    Twine(BlockIndex) +
+                                    " has no register definitions array");
 
+  size_t RegisterLoopIndex = 0;
   for (const auto &RegisterDefinitionValue : *RegisterDefinitions) {
```

Comment:
Same comment as below for the blocks: wouldn't an index-based for loop be simpler here?